### DISCUSSION TITLE
OSError: [Errno 36] File name too long

### DISCUSSION THREAD

**Discussion by deemok7** [talmolab/sleap#1354] [Help!]
Hi
I run this command
cmd = f"sleap-train {instantModifiedPath} {slpPath} --video-paths {videoPath}  --run_name {baseName} --save_viz --first-gpu "

I have more than 100 videos file that i listed in videoPath variable.

and get error like this:
Traceback (most recent call last):
  File "SleapTrainSlp_fromCmdLine.py", line 87, in <module>
    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/python/3.8.10/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/python/3.8.10/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 36] File name too long: 'sleap-train output/single_instance.json inputData/slpFiles/merged__F.slp --video-paths inputData/videoFiles/01.mp4 ..................... /99.mp4   --run_name merged__F --save_viz --first-gpu '

how can I solve it

**Comment by roomrys** [comment]
Hi @mmmahdiii,

It doesn't seem like your subprocess command even reaches the sleap code. Have you installed sleap in the environment where you are running your script?

Thanks,
Liezl

**Reply by deemok7** [reply]
Hi @roomrys 
I have Sleap installed in my environment. To verify it, I ran "sleap-train -h".
![image](https://github.com/talmolab/sleap/assets/84228651/4b657a73-c9b3-4200-954b-28014602a0e2)

The problem is that I have more than a hundred video files that I need to add to the "--video-paths" parameter, which is making my command too long. Therefore, I am looking for a solution to obtain a folder path instead of adding all individual video paths.

Thanks,
Mahdi

**Comment by roomrys** [comment]
Hi @mmmahdiii,

Have you tried just listing directories in the `--video_search_paths`? From the looks of the [`video_callback` function](https://github.com/talmolab/sleap/blob/1a0404c0ffae7b248eb360562b0bb95a42a287b6/sleap/io/dataset.py#L2572) and `find_path_using_paths`, specifying just the directories should work:
https://github.com/talmolab/sleap/blob/1a0404c0ffae7b248eb360562b0bb95a42a287b6/sleap/io/dataset.py#L2658-L2689

Let me know how it goes.

Thanks,
Liezl

**Reply by deemok7** [reply]
hi @roomrys 

I use 'sleep_train' as a command in Python code and set '--video_paths' as input. However, it can't accept a folder as input.
